cluster.go

```diff
@@ -1003,6 +1006,21 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 			continue
 		}
 
+		if !currentPin.PinOptions.ExpireAt.IsZero() && currentPin.PinOptions.ExpireAt.Before(timeNow) {
+			isClosest, err := c.isClosest(ctx, pCid)
+			if err != nil {
+				logger.Warning(err)
+			}
+			if isClosest {
+				logger.Infof("StateSync: unpinning %s, pin expired at %s", pCid, currentPin.PinOptions.ExpireAt.Format(time.RFC3339))
+				_, err := c.Unpin(ctx, pCid)
+				if err != nil {
+					logger.Warning(err)
+				}
+			}
+			continue
+		}
+
```

Comment:
This needs some sort of refactoring, it's too complex to read, and I think buggy (I think  a `continue` statement is missing after the warning?).